src/code/code_800E4FE0.c

```diff
@@ -177,12 +177,12 @@ AudioTask* func_800E5000(void) {
     task = &gAudioContext.currTask->task.t;
     task->type = M_AUDTASK;
     task->flags = 0;
-    task->ucode_boot = D_801120C0;
-    task->ucode_boot_size = 0x1000;
-    task->ucode_data_size = ((rspAspMainDataEnd - rspAspMainDataStart) * sizeof(u64)) - 1;
-    task->ucode = D_801120C0;
-    task->ucode_data = rspAspMainDataStart;
-    task->ucode_size = 0x1000;
+    task->ucode_boot = aspMainTextStart;
+    task->ucode_boot_size = SP_UCODE_SIZE;
+    task->ucode_data_size = (size_t)(aspMainDataEnd - aspMainDataStart) * sizeof(u64) - 1;
```

Comment:
unrelated to the PR but any idea of why there's a `- 1` ?

Also is this microcode different from the `naudio` ucode described in the manual

Reply:
The audio microcode is weird in that it doesn't use `rspboot` as a bootstrap, it bootstraps itself. The `SP_RD_LEN` and `SP_WR_LEN` hardware registers require you to enter `len - 1` as the dma length (hardware optimization, since a transfer length of 0 doesn't make sense it can be omitted). Usually, `rspboot` will subtract 1 to hide this from the user, but since the audio microcode doesn't use `rspboot` it does it manually here to save instructions in the microcode itself.

This microcode is different from the public microcode associated with the `naudio` library, although iirc one is believed to be a fork of the other, if I'm remembering discussions about sm64 audio correctly.